src/github.com/getlantern/flashlight/ui/ui.go

```diff
@@ -92,15 +92,25 @@ func Start(addr string) error {
 // Show opens the UI in a browser. It will wait for the UI addr come up for at most 3 seconds
 func Show() {
 	go func() {
-		addr, _ := url.Parse(uiaddr)
-		if err := waitforserver.WaitForServer("tcp", addr.Host, 3*time.Second); err != nil {
-			log.Error(err)
+		addr, er := url.Parse(uiaddr)
+		if er != nil {
+			log.Errorf("Could not parse url `%v` with error `%v`", uiaddr, er)
 			return
 		}
-		open.Run(uiaddr)
+		if err := waitforserver.WaitForServer("tcp", addr.Host, 10*time.Second); err != nil {
```

Comment:
Yeah, I basically just don't see any advantage to making it too short. If there's some unexpected delay with the the proxy starting, this will at least allow Lantern to still run.